### DISCUSSION TITLE
Condition on optional field

### DISCUSSION THREAD

**Discussion by MichaelStetner** [cue-lang/cue#3360] [Q&A] [answered]
I'm trying to validate some YAML that looks like this:

```
person:
  name: Sam
  employed: true
  jobtitle: Engineer
```

The rules are:
- "jobtitle" is only allowed if "employed" is true.
- "employed" is optional
- if "employed" is absent, treat it the same as "employed: false"

I tried writing the following cue to validate it:
```
person: closed({
    name: string
    employed?: bool
    if employed == true {
        jobtitle: string
    }
})
```

This works fine as long as "employed" is present. So the example at the top validates, and so does this:
```
person:
  name: Sam
  employed: false
```

But I get an error when I try to validate this:
```
person:
  name: Sam
```

The error message from `cue vet` is 
```
person: cannot reference optional field: employed:
    ./test.cue:4:8
```

Is there something I could write in cue to make this validate?

**Comment by slewiskelly** [comment, marked as answer]
Either of the following should work:

```cue
#Person: {
	name: string
} | {
	name:     string
	employed: bool

	if employed == true {
		jobtitle: string
	}
}
```

```cue
#Person: {
	name:      string
	employed?: bool

	if employed != _|_ {
		if employed == true {
			jobtitle: string
		}
	}
}
```

**Reply by MichaelStetner** [reply]
Thanks! That works.

Just for my understanding, why does this fail?
```
if employed != _|_ && employed == true 
```

(My intuition was that the logical operator && would be equivalent to nesting if statements)
